tests/v2.test.js

```diff
@@ -0,0 +1,443 @@
+        '180',
+        '270',
+        '360',
+        '1',
+        '1.5',
+        '!0',
+        '!1',
+        '!1.5',
+        '!90',
+      ];
+
+      test.each(rotations.map(rotation => [rotation]))('%s', rotation => badRequest({ rotation }));
+    });
+
+    describe('Invalid', () => {
+      const rotations = [
+        '-0',
+        '-90',
+        'foo',
+      ];
+
+      test.each(rotations.map(rotation => [rotation]))('%s', rotation => badRequest({ rotation }));
+    });
+  });
+
+  describe('Quality', () => {
+    describe('Supported', () => {
+      const qualities = [
+        'default',
+      ];
+
+      test.each(qualities.map(quality => [quality]))('%s', quality => ok({ quality }));
+    });
+
+    describe('Unsupported', () => {
+      const qualities = [
+        'bitonal',
+        'color',
+        'gray',
+      ];
+
+      test.each(qualities.map(quality => [quality]))('%s', quality => badRequest({ quality }));
+    });
+
+    describe('Invalid', () => {
+      const qualities = [
+        'foo',
+      ];
+
+      test.each(qualities.map(quality => [quality]))('%s', quality => badRequest({ quality }));
+    });
+  });
+
+  describe('Format', () => {
+    describe('Supported', () => {
+      const formats = {
+        jpg: 'pjpg',
+      };
+
+      test.each(Object.entries(formats))('%s', (iiifFormat, ioFormat) => ok({ format: iiifFormat }, { format: ioFormat }));
+    });
+
+    describe('Unsupported', () => {
+      const formats = [
+        'gif',
+        'jp2',
+        'pdf',
+        'png',
+        'tif',
+        'webp',
+      ];
+
+      test.each(formats.map(format => [format]))('%s', format => badRequest({ format }));
+    });
+
+    describe('Invalid', () => {
+      const formats = [
+        'foo',
+      ];
+
+      test.each(formats.map(format => [format]))('%s', format => badRequest({ format }));
+    });
+  });
+});
+
+describe('Info request', () => {
+  test.each([
+    [
+      '',
+      'pug-life.jpg',
+      {
+        width: 2000,
+        height: 1333,
+      },
+    ],
+    [
+      '/kittehs',
+      'pop6.jpg',
+      {
+        width: 4288,
+        height: 2848,
+      },
+    ],
+    [
+      '/kittehs',
+      'more%2Fcat-manipulating.jpg',
+      {
+        width: 3264,
+        height: 2448,
+      },
+    ],
+  ])('%s/%s/info.json', (prefix, identifier, requiredJson) => {
+    const json = Object.assign({
+      '@context': 'http://iiif.io/api/image/2/context.json',
+      '@id': baseUrl.concat(prefix, '/', identifier),
+      protocol: 'http://iiif.io/api/image',
+      profile: [
+        'http://iiif.io/api/image/2/level0.json',
+        {
+          formats: [
+            'jpg',
+          ],
+        },
+      ],
+    }, requiredJson);
+
+    expect.assertions(2);
+
+    return http.get(`${prefix}/${identifier}/info.json`)
+      .then((response) => {
+        expect(response.statusCode).toBe(200);
+        expect(JSON.parse(response.body)).toEqual(json);
+      });
```

Comment:
This is fine but I would favour `await`ing the request to keep the code flatter. 

Also although `expect.assertions()` (or `expect.hasAssertions()`) is essential if you are expecting in a `catch`, it's not necessary in a `then` because the only reason those expectations would not be reached is if an exception (or promise rejection) was thrown in which case the test would fail anyway.

Reply:
Makes sense. Tempted to use `simple: false` to avoid `catch`es too.

Reply:
Also, you probably already know this but if you want to change to `await` you need to make the test function `async`.